contracts/poap-manager/src/contract.rs

```diff
@@ -0,0 +1,493 @@
+        let env = mock_env();
+        let result = reply(
+            deps.as_mut(),
+            env,
+            Reply {
+                id: 2,
+                result: SubMsgResult::Ok(SubMsgResponse {
+                    events: vec![],
+                    data: None,
+                }),
+            },
+        );
+        assert!(result.is_err())
+    }
+
+    #[test]
+    fn poap_instantiate_with_invalid_instantiate_msg_error() {
+        let mut deps = mock_dependencies();
+        do_instantiate(deps.as_mut());
+        let env = mock_env();
+        let result = reply(
+            deps.as_mut(),
+            env,
+            Reply {
+                id: 1,
+                result: SubMsgResult::Ok(SubMsgResponse {
+                    events: vec![],
+                    data: None,
+                }),
+            },
+        );
+        assert!(result.is_err())
+    }
+
+    #[test]
+    fn poap_instantiate_reply_properly() {
+        let mut deps = mock_dependencies();
+        do_instantiate(deps.as_mut());
+        let env = mock_env();
+        let result = reply(deps.as_mut(), env, get_valid_instantiate_reply());
+        assert!(result.is_ok());
+        let address = POAP_ADDRESS.load(&deps.storage).unwrap();
+        let expected = Addr::unchecked("poap_contract_address");
+        assert_eq!(address, expected);
+    }
+
+    #[test]
+    fn claim_with_unsupported_desmos_deps_error() {
+        let mut deps = mock_dependencies();
+        do_instantiate(deps.as_mut());
+        let env = mock_env();
+        let info = mock_info(CREATOR, &vec![]);
+        let msg = ExecuteMsg::Claim {};
+        assert!(execute(deps.as_mut(), env, info, msg).is_err())
+    }
+
+    #[test]
+    fn claim_properly() {
+        let mut deps = mock_dependencies_with_custom_querier(&vec![]);
+        do_instantiate(deps.as_mut());
+        do_reply(deps.as_mut());
+        let env = mock_env();
+        let info = mock_info(CREATOR, &vec![]);
+        let msg = ExecuteMsg::Claim {};
+        assert!(execute(deps.as_mut(), env, info, msg).is_ok())
+    }
+
+    #[test]
+    fn mint_into_with_invalid_recipient_error() {
+        let mut deps = mock_dependencies();
+        do_instantiate(deps.as_mut());
+        do_reply(deps.as_mut());
+        let env = mock_env();
+        let info = mock_info(CREATOR, &vec![]);
+        let msg = ExecuteMsg::MintTo {
+            recipient: "a".into(),
+        };
+        assert!(execute(deps.as_mut(), env, info, msg).is_err())
+    }
+
+    #[test]
+    fn mint_into_properly() {
+        let mut deps = mock_dependencies();
+        do_instantiate(deps.as_mut());
+        do_reply(deps.as_mut());
+        let env = mock_env();
+        let info = mock_info(CREATOR, &vec![]);
+        let msg = ExecuteMsg::MintTo {
+            recipient: CREATOR.into(),
+        };
+        assert!(execute(deps.as_mut(), env, info, msg).is_ok())
+    }
+
+    #[test]
+    fn update_admin_with_invalid_msg_error() {
```

Comment:
Can we also add a test where a sender different than the admin try to update the admin?